### PR TITLE
Make number nullable

### LINKED ISSUE #164: Number in NumberPropertyValue needs to be Nullable.
As per  2021-08-16 Notion API version, empty database properties are now returned as null.

https://developers.notion.com/changelog/notion-version-2021-08-16#empty-database-properties-are-now-returned-as-null

I got this exception when querying and a record hand an empty Number type field.
Exception : Error converting value {null} to type 'System.Double'. Path 'number', line 1, position 1237.

I did not check other property types that may now be nullable.

### PR DESCRIPTION
**Pull request by kashifsoofi**
## Description

Make number property nullable.

Fixes #164 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] All unit tests passing

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by KoditkarVedant — CHANGES_REQUESTED** [review]
@kashifsoofi First of all thank you for your contribution. You also need to make `date`, `email`, and `rollup` types nullable.  You can find them in their respective `PropertyValue` type classes.

**Comment by kashifsoofi** [comment]
Sure I think i checked those, let me update, thanks

**Comment by kashifsoofi** [comment]
Email is string so that is nullable and so are the objects

File: Src/Notion.Client/Models/PropertyValue/DatePropertyValue.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class DatePropertyValue : PropertyValue
         /// Date
         /// </summary>
         [JsonProperty("date")]
-        public Date Date { get; set; }
+        public Date? Date { get; set; }
```
Comment (KoditkarVedant) [inline Src/Notion.Client/Models/PropertyValue/DatePropertyValue.cs:17]: @kashifsoofi you can remove the nullable as it is a class. rest LGTM.

Comment (kashifsoofi) [reply Src/Notion.Client/Models/PropertyValue/DatePropertyValue.cs]: Good spot, done

**Review by KoditkarVedant — APPROVED** [review]
LGTM.

**Pull request merged**